SignalMessaging/categories/Theme.h

```diff
@@ -56,4 +57,12 @@ extern NSString *const ThemeDidChangeNotification;
 
 @end
 
+#pragma mark -
+
+@interface OWSConversationColors (Theme)
+
+@property (nonatomic, readonly) UIColor *themeColor;
```

Comment:
I think `OWSConversationColors` will make for more readable code.

This category lets us pull out the theme-specific functionality.

Reply:
I like that you've pulled these into an object - though I'm not crazy about the name. I try to avoid having a single object expressed as a plural because inevitably, what do you call more than one of these? It's just a little readability/predictability hurdle. Would OWSConversationPalette be acceptable?

Also - what was the reasoning behind having a separate category? 

I pretty much always put things into the primary interface declaration unless:

1. I'm adding methods to classes I don't control (NSString+OWS) 
2. Or if I *privately* want to access some extra functionality on a class.
3. If I want to add Swift features to an Objc class without re-writing the entire class, I'll  create a swift extension of the class.

Reply:
> Would OWSConversationPalette be acceptable?

I think I'm going to go with `OWSConversationColor` (singular) since "conversation colors" is the term we're using within the team, in the UI, etc.

> Also - what was the reasoning behind having a separate category?

I was trying to avoid having `UIColor+OWS` depend on `Theme`, as you asked.  

https://github.com/signalapp/Signal-iOS/pull/3902#discussion_r220323636

I'll merge this category into the class.

Reply:
Since theme depends on UIColor, I do think it would be good for UIColor not to depend on Theme. 

In my experience, having strictly hierarchical dependencies causes less problems on average than things with circular dependencies. Splitting the circular dependency into a category does not remove the dependency, it just occludes it.

It's not always possible/feasible, but here it seems pretty arbitrary that the ConversationColor getters live on UIColor. e.g. why not ConversationColor.forColorName(String) (e.g.) rather than a map on UIColor?

I don't beat this to death... it's fine to leave the circular dependency as is, but I wanted to explain my rationale, since your usage of a category here, apparently only to satisfy me, tells me we had a misunderstanding.

Reply:
I see.  Is this what you have in mind?

https://github.com/signalapp/Signal-iOS/pull/3921